### DISCUSSION TITLE
Custom Panel Options

### DISCUSSION THREAD

**Discussion by alexk101** [imohitmayank/jaal#22] [General]
Hello,
I have been looking around for a good network visualization dashboard for a network simulation project that my team is developing, and after finding visdcc, I stumbled upon this project. Though it is very clean and has given me some basic functionality very quickly, I was wondering whether or not there was any good way to extend it's functionality, as from the documentation, the scope of the features is pretty barebones. In particular, I am looking for the ability to add in panels for the creation of new nodes in the network, load in data from json files, as well as a number of other panels that will interface with our simulation software to conduct tests on the constructed network. I know that this functionality can be implemented from the ground up using dash and visdcc, but I was just wondering if there was some way to extend the functionality of your project.

**Comment by imohitmayank** [comment]
Hi @alexk101, it seems you are planning to do a lot of modifications for your solution! 

That said, I think you can extend `jaal`. Even though `jaal` is not too extensive (the overall idea was to keep it simple), at the very least it provides you with some basic features and a structured package. For example, clear segregation of app layouts and backend between `layout.py` and `jaal.py`. Also, even in `jaal.py` we have clear segregation of callbacks.  In terms of adding new panels, you can cleanly segregate the layout and backend stuff and add them in the respective files for quick development. 

Best of luck, and do let me know in case of any confusion. 
Also, not sure if this is possible (due to NDAs etc), but would love to hear more or even see how your final solution (after all the modifications) looks like. :smile:

Cheers.
